ceno_zkvm/src/tables/ram/ram_circuit.rs

```diff
@@ -128,17 +128,20 @@ pub trait DynVolatileRamTable {
     const RAM_TYPE: RAMType;
     const V_LIMBS: usize;
 
-    const OFFSET_ADDR: Addr;
-    const END_ADDR: Addr;
+    // const OFFSET_ADDR: Addr;
+    // const END_ADDR: Addr;
+
+    fn offset_addr(platform: &Platform) -> Addr;
+    fn end_addr(platform: &Platform) -> Addr;
 
     fn name() -> &'static str;
 
-    fn max_len() -> usize {
-        (Self::END_ADDR - Self::OFFSET_ADDR) as usize / WORD_SIZE
+    fn max_len(platform: &Platform) -> usize {
+        (Self::end_addr(platform) - Self::offset_addr(platform)) as usize / WORD_SIZE
```

Comment:
This default body is not needed since it is overridden in the only implementation just above.